src/exception.jl

```diff
@@ -44,6 +49,28 @@ function show(io::IO, e::PyError)
     end
 end
 
+function pyerr_construct(msg::AbstractString)
+        ptype, pvalue, ptraceback = Ref{PyPtr}(), Ref{PyPtr}(), Ref{PyPtr}()
+        # equivalent of passing C pointers &exc[1], &exc[2], &exc[3]:
+        ccall((@pysym :PyErr_Fetch), Cvoid, (Ref{PyPtr},Ref{PyPtr},Ref{PyPtr}), ptype, pvalue, ptraceback)
+        ccall((@pysym :PyErr_NormalizeException), Cvoid, (Ref{PyPtr},Ref{PyPtr},Ref{PyPtr}), ptype, pvalue, ptraceback)
+        pyerr_construct(msg, PyObject(ptype[]), PyObject(pvalue[]), PyObject(ptraceback[]))
+end
+
+function pyerr_construct(msg::AbstractString, e::PyError)
+    pyerr_construct(msg, e.T, e.val, e.traceback)
+end
+
+function pyerr_construct(msg::AbstractString, ptype, pvalue, ptraceback)
```

Comment:
Probably we should declare `ptype`, `pvalue`, and `ptraceback` as `::PyObject` since those are the only correct types to pass here?